src/sidebar/store/test/use-store-test.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const { mount } = require('enzyme');
+const { createStore } = require('redux');
+const { createElement } = require('preact');
+const { act } = require('preact/test-utils');
+
+const useStore = require('../use-store');
+const { $imports } = useStore;
+
+const initialState = { value: 10, otherValue: 20 };
+const reducer = (state = initialState, action) => {
+  if (action.type === 'INCREMENT') {
+    return { ...state, value: state.value + 1 };
+  } else if (action.type === 'INCREMENT_OTHER') {
+    return { ...state, otherValue: state.otherValue + 1 };
+  } else {
+    return state;
+  }
+};
+
+describe('useStore', () => {
+  let renderCount;
+  let testStore;
+  let TestComponent;
+
+  beforeEach(() => {
+    renderCount = 0;
+
+    // eslint-disable-next-line react/display-name
+    TestComponent = () => {
+      renderCount += 1;
+      const aValue = useStore(store => store.getState().value);
+      return <div>{aValue}</div>;
+    };
+
+    testStore = createStore(reducer);
+    $imports.$mock({
+      '../util/service-context': {
+        useService: name => (name === 'store' ? testStore : null),
+      },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('returns result of `callback(store)`', () => {
+    const wrapper = mount(<TestComponent />);
+    assert.equal(wrapper.text(), '10');
+  });
+
+  it('re-renders when the store changes and result of `callback(store)` also changes', () => {
+    // An update which changes result of `callback(store)` should cause a re-render.
+    const wrapper = mount(<TestComponent />);
+    act(() => {
```

Comment:
I'm new to the `act` react test util, but I'll take React's word for it that it "makes your test run closer to how React works in the browser" (https://reactjs.org/docs/test-utils.html#act)

Reply:
That documentation is unfortunately not very clear. What `act(callback)` does in a nutshell is ensure that any state updates or effects scheduled during the callback get run before `act` returns. This is important because effects/state updates get debounced and run asynchronously (eg. if 10 state changes happen within 2ms of each other, the component gets re-rendered once shortly after the last change instead of 10 times).

A common pattern is something like this:

```js
const wrapper = mount(<MyWidget/>);
act(() => {
  // Do something which triggers an internal effect or state update in `MyWidget`.
  wrapper.find(ChildWidget).props().onClick();
});
wrapper.update(); // Update wrapper to reflect latest rendered output
assert(/* some assertion about the content of the wrapper */)
```